resources/vep/plugins/GADO.pm

```diff
@@ -0,0 +1,118 @@
+package GADO;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepPlugin);
+
+=head1 NAME
+ GADO
+=head1 SYNOPSIS
+ 
+=head1 DESCRIPTION
+ Plugin to annotate consequences with GADO scores modes based on their gene.
+=cut
+
+sub version {
+    return '0.1';
+}
+
+sub feature_types {
+    return ['Transcript'];
+}
+
+sub variant_feature_types {
+    return ['VariationFeature'];
+}
+
+my $self;
+
+sub get_header_info {
+    $self = GADO->new;
+    my $result;
+    $result->{GADO_SC} = "The combined prioritization GADO Z-score over the supplied HPO terms for this case.";
+    $result->{GADO_PD} = "The GADO predicion for the relation between the phenotypes and the gene, HC: high confidence, LC: low confidence.";
+    return $result;
+}
+
+sub new {
+    if (!(defined $self)) {
+        my $class = shift;
+        $self = $class->SUPER::new(@_);
+        my $gadoFile = $self->params->[0];
+        my $mappingFile = $self->params->[1];
+
+        die("ERROR: GADO output file not specified\n") unless $gadoFile;
+        die("ERROR: Gene mapping file not specified\n") unless $mappingFile;
+        my %gene_data = parseGadoFile($gadoFile);
+        my %gene_mapping = parseMappingFile($mappingFile);
+
+        $self->{gene_mapping} = \%gene_mapping;
+        $self->{gene_data} = \%gene_data;
+    }
+    return $self;
+}
+
+sub parseGadoFile {
+    my %gene_data;
+    my $file = $_[0];
+    open(FH, '<', $file) or die $!;
+
+    my @split;
+
+    while (<FH>) {
+        my $line = $_;
+        chomp($line);
+        @split = split(/\t/, $line);
+        $gene_data{$split[0]} = {Zscore => $split[3]};
+    }
+    return %gene_data
+}
+
+sub parseMappingFile {
+    my %mapping_data;
+    my $file = $_[0];
+    open(MAPPING_FH, '<', $file) or die $!;
+
+    my @split;
+
+    while (<MAPPING_FH>) {
+        my $line = $_;
+        chomp($line);
+        @split = split(/\t/, $line);
+        if (defined $split[0] and length $split[0]){
+            $mapping_data{$split[1]} = $split[0];
```

Comment:
1) Gene mapping not always possible, see for example https://github.com/molgenis/vip/blob/31f674bb924b0de38a2e5fea941d7c6d529efb57/resources/biomart_ensembl_entrez_mapping.txt#L35. Is that taken care of on L99-100?

2) how about logging a warning for unmappable genes?

3) in the HLHS run with the poc we noticed that many variant-effects with a HPO annotation unexpectedly did not get a GADO score. could that we related to a gene id mapping issue?

Reply:
in the PoC we did not map at all, but used the symbols